File: baba-chatbot-net/src/Baba.Chatbot.Application/Conversation/Guardrails/ContentModerationResult.cs
namespace Baba.Chatbot.Application.Conversation.Guardrails;

/// <summary>
/// Resultado de la moderación de contenido de mensajes de usuario
/// </summary>
public class ContentModerationResult
{
    public bool IsAppropriate { get; set; }
    public List<ModerationFlag> Flags { get; set; } = new();
    public string? SuggestedResponse { get; set; }
    public ModerationSeverity Severity { get; set; }
}

/// <summary>
/// Tipos de violaciones detectadas en la moderación de contenido
/// </summary>
public enum ModerationFlag
{
    None,
    OffTopic,
    HateSpeech,
    Violence,
    SexualContent,
    Harassment,
    PII,
    Spam,
    IllegalContent,
    ManipulationAttempt
}

/// <summary>
/// Nivel de severidad de una violación de moderación
/// </summary>
public enum ModerationSeverity
{
    None,
    Low,
    Medium,
    High,
    Critical
}

/// <summary>
/// Resultado de la validación de guardrails aplicada a respuestas del LLM
/// </summary>
public class GuardrailsValidationResult
{
    public bool IsValid { get; set; }
    public List<string> Violations { get; set; } = new();
    public string? CleanedContent { get; set; }
    public bool RequiresHumanEscalation { get; set; }
}

File: baba-chatbot-python/src/baba_chatbot/application/conversation/guardrails/content_moderation_result.py
"""
Resultados y enumeraciones para moderación de contenido y validación de guardrails.
"""
from enum import Enum
from typing import List
from dataclasses import dataclass, field


class ModerationFlag(Enum):
    """Tipos de violaciones detectadas por el sistema de moderación."""
    HATE_SPEECH = "hate_speech"
    VIOLENCE = "violence"
    SEXUAL_CONTENT = "sexual_content"
    HARASSMENT = "harassment"
    OFF_TOPIC = "off_topic"


class ModerationSeverity(Enum):
    """Niveles de severidad de las violaciones."""
    NONE = 0
    LOW = 1
    MEDIUM = 2
    HIGH = 3


@dataclass
class ContentModerationResult:
    """
    Resultado de la moderación de contenido de un mensaje.
    """
    is_appropriate: bool = True
    severity: ModerationSeverity = ModerationSeverity.NONE
    flags: List[ModerationFlag] = field(default_factory=list)
    suggested_response: str = ""

    def __post_init__(self):
        """Asegura que flags sea una lista mutable."""
        if not isinstance(self.flags, list):
            self.flags = list(self.flags) if self.flags else []


@dataclass
class GuardrailsValidationResult:
    """
    Resultado de la validación de guardrails de una respuesta del LLM.
    """
    is_valid: bool = True
    violations: List[str] = field(default_factory=list)
    cleaned_content: str = ""
    requires_human_escalation: bool = False

    def __post_init__(self):
        """Asegura que violations sea una lista mutable."""
        if not isinstance(self.violations, list):
            self.violations = list(self.violations) if self.violations else []

